src/components/time/time.stories.tsx

```diff
@@ -0,0 +1,384 @@
+};
+
+InputHint.parameters = {
+  chromatic: { disableSnapshot: false },
+  themeProvider: { chromatic: { theme: "sage" } },
+};
+
+export const Required: ComponentStory<typeof Time> = () => {
+  const [value, setValue] = useState<TimeValue>({
+    hours: "",
+    minutes: "",
+  });
+
+  const handleChange = (ev: TimeInputEvent) => {
+    setValue(ev.target.value);
+  };
+
+  return (
+    <Box p={2}>
+      <Time required value={value} onChange={handleChange} label="Time" />
+    </Box>
+  );
+};
+
+Required.parameters = {
+  chromatic: { disableSnapshot: false },
+  themeProvider: { chromatic: { theme: "sage" } },
+};
+
+export const IsOptional: ComponentStory<typeof Time> = () => {
+  const [value, setValue] = useState<TimeValue>({
+    hours: "",
+    minutes: "",
+  });
+
+  const handleChange = (ev: TimeInputEvent) => {
+    setValue(ev.target.value);
+  };
+
+  return (
+    <Box p={2}>
+      <Time isOptional value={value} onChange={handleChange} label="Time" />
+    </Box>
+  );
+};
+
+IsOptional.parameters = {
+  chromatic: { disableSnapshot: false },
+  themeProvider: { chromatic: { theme: "sage" } },
+};
+
+export const Disabled: ComponentStory<typeof Time> = () => {
+  const [value, setValue] = useState<TimeValue>({
+    hours: "",
+    minutes: "",
+    period: "AM",
+  });
+
+  const handleChange = (ev: TimeInputEvent) => {
+    setValue(ev.target.value);
+  };
+
+  return (
+    <Box p={2}>
+      <Time
+        value={value}
+        onChange={handleChange}
+        label="Time"
+        inputHint="Hint text"
+        disabled
+      />
+    </Box>
+  );
+};
+
+Disabled.parameters = {
+  chromatic: { disableSnapshot: false },
+  themeProvider: { chromatic: { theme: "sage" } },
+};
+
+export const ReadOnly: ComponentStory<typeof Time> = () => {
+  const [value, setValue] = useState<TimeValue>({
+    hours: "",
+    minutes: "",
+    period: "AM",
+  });
+
+  const handleChange = (ev: TimeInputEvent) => {
+    setValue(ev.target.value);
+  };
+
+  return (
+    <Box p={2}>
+      <Time
+        value={value}
+        onChange={handleChange}
+        label="Time"
+        inputHint="Hint text"
+        readOnly
+      />
+    </Box>
+  );
+};
+
+ReadOnly.parameters = {
+  chromatic: { disableSnapshot: false },
+  themeProvider: { chromatic: { theme: "sage" } },
+};
+
+export const Sizes: ComponentStory<typeof Time> = () => {
+  const [value, setValue] = useState<{
+    small: TimeValue;
+    medium: TimeValue;
+    large: TimeValue;
+  }>({
+    small: {
+      hours: "",
+      minutes: "",
+      period: "AM",
+    },
+    medium: {
+      hours: "",
+      minutes: "",
+      period: "AM",
+    },
+    large: {
+      hours: "",
+      minutes: "",
+      period: "AM",
+    },
+  });
+
+  const handleChange = (
+    ev: TimeInputEvent,
+    size: "small" | "medium" | "large"
+  ) => {
+    setValue((p) => ({
+      ...p,
+      [size]: ev.target.value,
+    }));
+  };
+
+  return (
+    <Box p={2}>
+      {SIZES.map((size) => (
```

Comment:
suggestion(non-blocking): `SIZES` won't appear in the code block of the story so should be moved into the story

I know it's better to not have to keep setting `SIZES` but it doesn't have the context for the user. I do think we need a nicer way to show our code blocks personally though, but we've discussed making improvements to them in general so it would be part of that

Reply:
yeah valid point I'll remove the map here and just go with an individual story for each

Reply:
Much better